src/main/java/org/kitteh/irc/client/library/Client.java

```diff
@@ -98,26 +98,233 @@
      */
     interface Builder {
         /**
-         * Sets the supplier of the actor tracker.
-         * <p>
-         * By default, the {@link DefaultActorTracker} is used.
-         *
-         * @param supplier supplier
-         * @return this builder
-         * @see DefaultActorTracker
-         */
-        @NonNull Builder actorTracker(@NonNull Function<Client.WithManagement, ? extends ActorTracker> supplier);
+         * Listener builder methods.
+         */
+        interface Listeners {
+            /**
+             * Sets a listener for all incoming messages from the server.
+             * <p>
+             * All messages are passed from a single, separate thread.
+             *
+             * @param listener input listener or null to not listen
+             * @return this builder
+             */
+            @NonNull Listeners input(@Nullable Consumer<String> listener);
+
+            /**
+             * Sets a listener for all outgoing messages to the server.
+             * <p>
+             * All messages are passed from a single, separate thread.
+             *
+             * @param listener output listener or null to not listen
+             * @return this builder
+             */
+            @NonNull Listeners output(@Nullable Consumer<String> listener);
+
+            /**
+             * Sets a listener for all thrown exceptions on this client. By default,
+             * a consumer exists which calls Throwable#printStackTrace() on all
+             * received exceptions.
+             * <p>
+             * All exceptions are passed from a single, separate thread.
+             *
+             * @param listener catcher of throwable objects or null to not listen
+             * @return this builder
+             */
+            @NonNull Listeners exception(@Nullable Consumer<Exception> listener);
+
+            /**
+             * Returns to the root builder.
+             *
+             * @return the root builder
+             */
+            @NonNull Builder then();
+        }
+
+        /**
+         * WebIRC builder.
+         */
+        interface WebIrc {
```

Comment:
Needs handling of partially null info, right? Before it was all or nothing, as [WEBIRC](https://ircv3.net/specs/extensions/webirc.html) requires them all.